File: src/main/webapp/js/controller/login-controller.js
appEntra21.controller("loginController", function($scope, $http, $window,
		$location) {

	$scope.usuario = {};
	$scope.mensagem = ""
	var urlApi = 'rest/login/autenticar/'

	$scope.autenticar = function() {

		$http.post(urlApi, $scope.usuario).then(
				function(response) {

					console.log(response.data);

					if (response.data.email == $scope.usuario.email
							&& response.data.senha == $scope.usuario.senha) {
						$location.path('despesas');

					}

				}, function(response) {
					console.log(response.status);
					if (response.status == '500') {
						alert("Verificar usuario ou senha")
						$scope.mensagem = "Verificar usuario ou senha"
					}
				});

		this.logout = function() {
			$rootScope.response.data = null;
			$location.path('indexview')
		}
	}
});

File: src/main/java/br/com/entra21/walletControl/controller/LoginController.java
package br.com.entra21.walletControl.controller;

import java.util.logging.Level;
import java.util.logging.Logger;

import javax.ws.rs.Consumes;
import javax.ws.rs.POST;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.WebApplicationException;
import javax.ws.rs.core.MediaType;
import javax.ws.rs.core.Response;

import br.com.entra21.walletControl.dao.LoginDAO;
import br.com.entra21.walletControl.model.Usuario;

@Path("login/")
public class LoginController {

	@POST
	@Consumes(MediaType.APPLICATION_JSON)
	@Produces(MediaType.APPLICATION_JSON)
	@Path("autenticar/")
	public Usuario auth(Usuario usuario) throws WebApplicationException {
		try {

			if (usuario.getEmail() == null || usuario.getSenha() == null) {
				throw new WebApplicationException("Login e senha obrigatorios");
			}
			LoginDAO usuAutenticar = new LoginDAO();
			Usuario usuarioAtenticado = usuAutenticar.autenticarLogin(usuario);

			if (usuarioAtenticado == null) {

				throw new WebApplicationException("Usuario não encontrado");
			}

			if (!usuarioAtenticado.getSenha().equals(usuario.getSenha())) {
				throw new WebApplicationException("Usuario ou senha inválido");
			}

			return usuarioAtenticado;
		} catch (Exception ex) {
			Logger.getLogger(UsuarioController.class.getName()).log(Level.SEVERE, null, ex);
			throw new WebApplicationException(Response.Status.INTERNAL_SERVER_ERROR);
		}
	}

	private class LoginResponse {

		public String token;

		public LoginResponse(String token) {
			this.token = token;
		}
	}

}
